packages/icon/package.py

```diff
@@ -1,11 +1,20 @@
-import os, subprocess, glob
+import os, subprocess, glob, re
 from collections import defaultdict
 
 from llnl.util import lang, filesystem, tty
 from spack.util.environment import is_system_path, dump_environment
 from spack.util.executable import which_string, which
 
 
+def check_variant_fcgroup(fcgroup):
+    pattern = re.compile(r"^[A-Z]+;.+;.")
```

Comment:
would this regex catch the error is the flags would have a space: e.g. "-O2 -g" since you say this is not supported

Reply:
This does not even work, because you cannot have a string including whitespaces enclosed by "" in spack.

Spack treats -g in this case as a separate variant, what fails.